pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -621,70 +627,46 @@ private void resetCursor(Position finalPosition, CompletableFuture<Void> future)
             return;
         }
 
-        final CompletableFuture<Void> disconnectFuture;
-
-        // Lock the Subscription object before locking the Dispatcher object to avoid deadlocks
-        synchronized (this) {
-            if (dispatcher != null && dispatcher.isConsumerConnected()) {
-                disconnectFuture = dispatcher.disconnectAllConsumers();
-            } else {
-                disconnectFuture = CompletableFuture.completedFuture(null);
-            }
+        if (dispatcher != null) {
```

Comment:
Mostly, I started pulling the thread and changing the seek behavior to avoid closing consumers in between. I'll pull that change out (in different PR) and apply it before this PR.